### ISSUE TITLE
Changing the Value of an Initializer

### ISSUE THREAD

**Issue by sfrooster** [dsherret/ts-simple-ast#38]
I have a line in a class in a file I'm processing like this:

```typescript
public static readonly Version: string = "1.0"; 
```
I get via:
```typescript
ClassDeclaration.getStaticProperties
```
and I want to change it to
```typescript
public static readonly Version: string = "2.0"; 
```

The call I make is 
```typescript
setInitializer("\"2.0\"");
```

But it blows up with:
```
Code:
        public static readonly Version: string; = "2.0"   // version to pass to EDAProperties
    at BaseError (/Users/mhaan/AppDev/tscompile/node_modules/ts-simple-ast/dist/errors/BaseError.js:5:9)
    at InvalidOperationError (/Users/mhaan/AppDev/tscompile/node_modules/ts-simple-ast/dist/errors/InvalidOperationError.js:6:9)
    at handleNode (/Users/mhaan/AppDev/tscompile/node_modules/ts-simple-ast/dist/manipulation/insertStraight.js:28:19)
    at handleNode (/Users/mhaan/AppDev/tscompile/node_modules/ts-simple-ast/dist/manipulation/insertStraight.js:37:13)
    at handleNode (/Users/mhaan/AppDev/tscompile/node_modules/ts-simple-ast/dist/manipulation/insertStraight.js:37:13)
    at handleNode (/Users/mhaan/AppDev/tscompile/node_modules/ts-simple-ast/dist/manipulation/insertStraight.js:37:13)
    at handleNode (/Users/mhaan/AppDev/tscompile/node_modules/ts-simple-ast/dist/manipulation/insertStraight.js:37:13)
    at handleNode (/Users/mhaan/AppDev/tscompile/node_modules/ts-simple-ast/dist/manipulation/insertStraight.js:37:13)
    at handleNode (/Users/mhaan/AppDev/tscompile/node_modules/ts-simple-ast/dist/manipulation/insertStraight.js:37:13)
    at handleNode (/Users/mhaan/AppDev/tscompile/node_modules/ts-simple-ast/dist/manipulation/insertStraight.js:37:13)
```
setInitializer first removes the existing initializer bringing the semi-colon all the way over, then adds the new initializer which ends-up being after the semi-colon.

**Comment by sfrooster** [comment]
This isn't exactly right as a fix (because, for example, a comment could be placed on the line following the statement), but this does approximate a solution:

```typescript
        setInitializer(text: string) {
            errors.throwIfNotStringOrWhitespace(text, nameof(text));

            if (this.hasInitializer())
                this.removeInitializer();

            insertStraight({
                insertPos: this.getFullText().endsWith(";") ? this.getEnd() - 1 : this.getEnd(),
                parent: this,
                newCode: ` = ${text}`
            });
            return this;
        }
```

**Comment by dsherret** [comment]
Sorry for the long reply. I had a lot going on after work this week.

This is fixed in 0.30.1.  The trick here is to insert at the start position of the semi-colon when it exists:

```typescript
function getInsertPos(node: Node) {
    const lastChild = node.getLastChild();
    if (lastChild != null && lastChild.getKind() === ts.SyntaxKind.SemicolonToken)
        return lastChild.getPos();
    return node.getEnd();
}
```

That will handle calling `setInitializer` when there's a comment. For example, `class Identifier { prop/*comment*/; }` will change it to `class Identifier { prop = 2/*comment*/; }`.

**Comment by dsherret** [comment]
The reason this bug occurred is because I had only tested this with enum members. I think at the time I added this I didn't have support for classes yet.
